mpire/comms.py

```diff
@@ -611,13 +620,17 @@ def drain_and_join_queue(self, q: mp.JoinableQueue, join: bool = True) -> None:
         :param q: Queue to join
         :param join: Whether to join the queue or not
         """
-        process = mp.Process(target=self._drain_and_join_queue, args=(q, join))
-        process.start()
-        process.join(timeout=5)
-        if process.is_alive():
-            logger.debug("Draining queue failed, skipping")
-            process.terminate()
-            process.join()
+        try:
+            process = mp.Process(target=self._drain_and_join_queue, args=(q, join))
+            process.start()
+            process.join(timeout=5)
+            if process.is_alive():
+                logger.debug("Draining queue failed, skipping")
+                process.terminate()
+                process.join()
+        except (OSError, RuntimeError):
```

Comment:
So basically windows throws an error here, but nothing needs to be done? Does everything work well down the pipeline from this moment? What does it crash on exactly?

Reply:
This part is only triggered when an exception occurs.

Windows exception handling is pretty bad as Windows has very limited support for signals. Therefore, workers are often terminated instead of gracefully killed, which results in queues being corrupt. As a result this process refuses to start. This only happens on Windows and AFAIK you can still continue working as normal. The exception is correctly raised and the traceback is ok.